lib/ansible/module_utils/facts/virtual/linux.py

```diff
@@ -68,6 +68,22 @@ def get_virtual_facts(self):
                         virtual_facts['virtualization_role'] = 'guest'
                         found_virt = True
 
+        # the lxd communication sockets are good indicators for lxc guest/host
+        # https://lxd.readthedocs.io/en/latest/dev-lxd/
+        if os.path.exists('/dev/lxd/sock'):
+            guest_tech.add('lxc')
+            if not found_virt:
```

Comment:
To keep backwards compatibility for the virtualization_type and virtualization_role facts, this would need to occur at the end of the function. First found wins, so this would take precedence over any of the facts found later.